### DISCUSSION TITLE
How make it Tauri

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#5576] [Ideas]
Hello !
i want to convert code server to Tauri application !Where i can found something like html of the code server or vscode to make it somthing like vscode in electron ?

**Comment by jsjoeio** [comment]
I'm not super familiar with how Tauri works or how difficult this would be. I found these resources:
- https://tauri.app/v1/guides/getting-started/setup/integrate
- https://replit.com/talk/learn/Convert-your-web-project-to-a-desktop-app-with-Tauri/143553
- https://github.com/tauri-apps/tauri/issues/986

code-server uses vscode's html file for loading the IDE which is this: https://github.com/microsoft/vscode/blob/a44493aed3c9ccdec18a4bc00869403736b29c64/src/vs/code/browser/workbench/workbench.html

Feel free to use this thread as a rubber duck and we can chime in here and there to help you figure this out!

**Reply by lg-nineoneninethreenine** [reply]
I know a buddy who switched form Electron to Tauri via TS ofc, you want me to chime him in with his consent if you need help? He did the switch mostly by himself and his friend, but it(Tauri) does work better and faster than Electron's framework sometimes.

**Reply by jsjoeio** [reply]
Yeah that would be helpful!

**Comment by ghost** [comment]
I need only html css and javascript of vscode tanks

**Reply by lg-nineoneninethreenine** [reply]
Plot twist: VS Code, and therefore code-server is built on TypeScript. If you need help or suggestions on that, I will chime my buddy in if it's ok with him. Mostly I think it(Tauri) needs a bit more code to make it an actual window, but IDK. If you want to figure it out, all the best, it would def help people who need a local app.
